src/qt/walletview.cpp

```diff
@@ -166,7 +169,12 @@ void WalletView::processNewTransaction(const QModelIndex& parent, int start, int
     QString address = ttm->data(index, TransactionTableModel::AddressRole).toString();
     QString label = ttm->data(index, TransactionTableModel::LabelRole).toString();
 
-    Q_EMIT incomingTransaction(date, walletModel->getOptionsModel()->getDisplayUnit(), amount, type, address, label);
+    QString wallet;
+    if (bitcoinGui->multiWallet()) {
+        wallet = walletModel->getWalletName();
+    }
+
+    Q_EMIT incomingTransaction(date, walletModel->getOptionsModel()->getDisplayUnit(), amount, type, address, label, wallet);
```

Comment:
Why not remove the `bitcoinGui` member and always send the wallet name?

Reply:
When not in multiwallet mode, this causes the "Wallet" line to be omitted from the notification popup.

Reply:
Since only `bitcoingui.cpp` listens to the `incomingTransaction` signal, it could be added or not added there. IMO the signal emitter should not decide wether to show the wallet name or not.